### ISSUE TITLE
[Phase 2] Spectral Masking Model

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/speech-enhancement-system#3]
## Phase 2: Spectral Masking Model

**Parent:** #1
**Depends on:** #2

### Objectives
Implement LSTM-based spectral masking network for speech enhancement.

### Tasks

- [ ] Implement `SpectralMaskNet` class
- [ ] Add encoder (Linear + LayerNorm + ReLU)
- [ ] Add bidirectional LSTM layers
- [ ] Add decoder with sigmoid output
- [ ] Verify output mask is in [0, 1]
- [ ] Test forward pass with batch input
- [ ] Check parameter count (<5M target)

### Files to Create/Modify

| File | Action | Description |
|------|--------|-------------|
| `src/models.py` | Create | Model architectures |

### Code Reference

```python
import torch.nn as nn

class SpectralMaskNet(nn.Module):
    def __init__(self, n_fft: int = 512, hidden_size: int = 512):
        super().__init__()
        input_size = n_fft // 2 + 1  # 257

        self.encoder = nn.Sequential(
            nn.Linear(input_size, hidden_size),
            nn.LayerNorm(hidden_size),
            nn.ReLU(),
            nn.Dropout(0.2)
        )

        self.lstm = nn.LSTM(
            hidden_size, hidden_size,
            num_layers=2,
            bidirectional=True,
            batch_first=True,
            dropout=0.2
        )

        self.decoder = nn.Sequential(
            nn.Linear(hidden_size * 2, hidden_size),
            nn.LayerNorm(hidden_size),
            nn.ReLU(),
            nn.Dropout(0.2),
            nn.Linear(hidden_size, input_size),
            nn.Sigmoid()
        )

    def forward(self, noisy_mag):
        # (batch, freq, time) -> (batch, time, freq)
        x = noisy_mag.transpose(1, 2)
        x = self.encoder(x)
        x, _ = self.lstm(x)
        mask = self.decoder(x)
        return mask.transpose(1, 2)
```

### Definition of Done

- [ ] Model instantiates without errors
- [ ] Forward pass produces correct output shape
- [ ] Mask values are in [0, 1] range
- [ ] Gradients flow correctly (no NaN)
- [ ] Parameter count printed and reasonable

### Technical Notes

**For junior developers:**
- LayerNorm helps stabilize LSTM training
- Bidirectional LSTM doubles hidden size in output
- Sigmoid ensures mask is in [0, 1] for multiplication
- Transpose needed because LSTM expects (batch, time, features)
